src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -359,7 +366,17 @@ internal static SafeSslHandle AllocateSslHandle(SafeFreeSslCredentials credentia
 
         internal static SecurityStatusPal SslRenegotiate(SafeSslHandle sslContext, out byte[]? outputBuffer)
         {
-            int ret = Interop.Ssl.SslRenegotiate(sslContext);
+            int ret;
+
+            // TODO: optimize this check
+            if (new SslConnectionInfo(sslContext).Protocol == (int) SslProtocols.Tls13)
+            {
+                ret = Interop.Ssl.SslVerifyClientPostHandshake(sslContext);
+            }
```

Comment:
I am thinking of moving this inside the `Ssl.SslRenegotiate` shim, inside OpenSSL, we can use `SSL_version` to check trivially.

Reply:
SSlStream.PalLinux may have access. or perhaps hide it in native C shim.

Reply:
Moved into Ssl.Renegotiate